specs/phase1/beacon-chain.md

```diff
@@ -928,13 +931,16 @@ def process_crosslinks(state: BeaconState,
             attestation for attestation in attestations
             if is_on_time_attestation(state, attestation) and attestation.data.index == committee_index
         ]
-        shard = compute_shard_from_committee_index(state, committee_index, on_time_attestation_slot)
-        winning_root = process_crosslink_for_shard(state, committee_index, shard_transitions[shard], shard_attestations)
-        if winning_root != Root():
-            # Mark relevant pending attestations as creating a successful crosslink
-            for pending_attestation in state.current_epoch_attestations:
-                if is_winning_attestation(state, pending_attestation, committee_index, winning_root):
-                    pending_attestation.crosslink_success = True
+        if len(shard_attestations) > 0:
```

Comment:
I think it reasonable to keep the `shard = ` computation as previous and not get `shard` from `attestation` in this case. This method is less about a single attestation (and it's data) and instead about a committee and it's shard. So calculating the committees shard from the helper instead of grabbing from a particular attestation seems safe and more readable

The previous method was more readable and same security so I'd lean in that direction.